wallet/dapps-communication.md

```diff
@@ -9,3 +9,353 @@ Initial meta issue: <https://github.com/massalabs/massa-standards/issues/13>
+
+```typescript
+import { providers, Provider, Wallet } from 'massa-wallet-provider';
+
+async function interactWithMassaWallet() {
+  // Get the available wallet providers
+  const availableProviders: Provider[] = await providers();
+
+  // Display the names of the available wallet providers on the website
+  const providerNames: string[] = availableProviders.map(provider => provider.name());
+  document.getElementById('provider-names').innerText = providerNames.join(', ');
+
+  // Allow the user to select a wallet provider
+  const selectedProvider: Provider = availableProviders[0];
+
+  // Display the wallets registered with the selected provider on the website
+  const wallets: Wallet[] = await selectedProvider.wallets();
+  const walletAddresses: string[] = await Promise.all(wallets.map(async wallet => wallet.address()));
+  document.getElementById('wallet-addresses').innerText = walletAddresses.join(', ');
+
+  // Allow the user to select a wallet
+  const selectedWallet: Wallet = wallets[0];
+
+  // Display the balance of the selected wallet on the website
+  const walletBalance: BigInt = await selectedWallet.balance();
+  document.getElementById('wallet-balance').innerText = walletBalance.toString();
+
+  // Allow the user to sign a transaction
+  const payload: Uint8Array = new Uint8Array([1, 2, 3]);
+  const [pubKey, signature]: [string, Uint8Array] = await selectedWallet.sign(payload);
+}
+```
+
+Bob can utilize our wallet-provider JS library, which provides the following functions:
+
+```typescript
+export async function providers(): Promise<Provider[]> { }
+
+export class Wallet {
+  async address(): Promise<string> { }
+
+  async balance(): Promise<BigInt> { }
+
+  async sign(payload: Uint8Array): Promise<[string, Uint8Array]> { }
+}
+
+export class Provider {
+  name(): string { }
+
+  async wallets(): Promise<Wallet[]> { }
+
+  async importWallet(): Promise<Wallet> { }
+
+  async deleteWallet(wallet: Wallet): Promise<void> { }
+}
+```
+
+### Alice, the browser extension developer
+
+Alice wants to create a browser extension that can interact with the Massa ecosystem using the wallet-provider JS library.
+However, Alice needs to interact with the library from the [content script](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts), which is separate from the webpage script.
+
+Alice needs the ability to perform the following tasks:
+
+- Register her wallet extension as a provider with the web page.
+- Receive commands from the web page and respond to them accordingly.
+
+To do this, she can use the Massa wallet-provider-content-script JS library:
+
+```typescript
+import {
+  registerAsMassaWalletProvider,
+  listWallet,
+  deleteWallet,
+  importWallet,
+  balance,
+  sign,
+} from 'wallet-provider-content-script';
+
+// Register Alice's wallet extension as a provider with the web page
+registerAsMassaWalletProvider('Alice Wallet').then((success) => {
```

Comment:
I don't understand how this function will save this callback and trigger it again when ask by the dapp builder

Reply:
See [ehre](https://github.com/massalabs/massa-standards/pull/27/commits/29d50a06b196fd93680af1e98d00c700346c5e51#diff-dad455c0814fa2229eaae244a61b26f2f312670c0d14b6da452cd46744739956R247-R260).